### PR TITLE
Generate json schemas for CRDs.

### LINKED ISSUE #1091: Generate json schemas for validation from FybrikApplication and FybrikModule code
Looking at the flow for taxonomy validation of CRDs (e.g., FybrikApplication and [FybrikModule](https://github.com/fybrik/fybrik/pull/804)) I'm a bit worried about keeping consistency between the structure in the CRD and the structure in the validation object. The latter must be a subset of the former with `additionalProperties: true` set in the right places to allow the additional fields that don't require validation. Whenever the structure of the CRD would change we will need to be careful about the structure of the validation object. 

_Originally posted by @roee88 in https://github.com/fybrik/fybrik/discussions/848_

### PR DESCRIPTION
**Pull request by Mohammad-nassar10**
Generate json schemas for `fybrikApplication` and `fybrikModule` automatically using an updated version of the `json-schema-generator`. Also, adding a test `fybrikmodule_webhook_test.go` for checking the `fybrikModule` taxonomy with valid and invalid `fybrikModules` 

Closes #1091 

Signed-off-by: nassarmu10 <mohammad.nassar@ibm.com>

### CODE REVIEW

**Comment by shlomitk1** [comment]
Why is this PR required? Please connect to the relevant issue.

**Review by shlomitk1 — COMMENTED** [review]
Please rebase. There are changes that are in master, yet marked as diffs here.

File: manager/controllers/app/fybrikmodule_controller.go [hunk 1]
```diff
@@ -107,7 +107,8 @@ func ValidateFybrikModule(module *fapp.FybrikModule, taxonomyFile string) error
 	var allErrs []*field.Error
 
 	// Convert Fybrik module Go struct to JSON
-	moduleJSON, err := json.Marshal(module)
+	spec := module.Spec.DeepCopy()
+	moduleJSON, err := json.Marshal(spec)
```
Comment (shlomitk1) [inline manager/controllers/app/fybrikmodule_controller.go:111]: Why do you need a deep copy here? json.Marshal(&module.Spec) should be enough, you are not using the saved pointer, just a list of bytes...

**Comment by roytman** [comment]
@Mohammad-nassar10 please rebase again

**Pull request merged**